Server/Plugins/APIDump/Classes/Geometry.lua

```diff
@@ -977,52 +977,103 @@ return
 	{
 		Desc = [[
 This class provides an easy-to-use interface for tracing lines through individual
-blocks in the world. It will call the provided callbacks according to what events it encounters along the
-way.</p>
-<p>
-For the Lua API, there's only one static function exported that takes all the parameters necessary to do
-the tracing. The Callbacks parameter is a table containing all the functions that will be called upon the
-various events. See below for further information.
+blocks in the world. It can either be used to call the provided callbacks according
+to what events it encounters along the way, or there are shortcut functions used for
+the most popular tracing reasons - line of sight and solid hits.
 		]],
 		Functions =
 		{
-			Trace =
+			FirstSolidHitTrace =
 			{
-				IsStatic = true,
-				Params =
 				{
+					IsStatic = true,
+					Params =
 					{
-						Name = "World",
-						Type = "cWorld",
+						{ Name = "World",  Type = "cWorld" },
+						{ Name = "StartX", Type = "number" },
+						{ Name = "StartY", Type = "number" },
+						{ Name = "StartZ", Type = "number" },
+						{ Name = "EndX",   Type = "number" },
+						{ Name = "EndY",   Type = "number" },
+						{ Name = "EndZ",   Type = "number" },
 					},
+					Returns =
 					{
-						Name = "Callbacks",
-						Type = "table",
+						{ Name = "HasHitSolid",    Type = "bool" },
+						{ Name = "HitCoords",      Type = "Vector3d" },
+						{ Name = "HitBlockCoords", Type = "Vector3i" },
+						{ Name = "HitBlockFace",   Type = "eBlockFace" },
```

Comment:
I'm not too happy with the names for these three return values (params in C++ code). If anyone can suggest a better name, I'm open to suggestions.
`HitCoords` are the exact coords of the hit
`HitBlockCoords` are the coords of the block that has been hit
`HitBlockFace` is the face of the block that has been hit.